docs/elasticsearch.rst

```diff
@@ -63,9 +65,15 @@ expected to return a ``doc_dict``, so just add the ``username`` to that.
 Building the new index
 ''''''''''''''''''''''
 Once you've made the change, you'll need to build a new index which uses
-that new mapping, so you'll have to update the hash at the top of the file.
-This can just be a random alphanumeric string.  This will trigger a preindex
-as outlined in the `Indexes` section.
+that new mapping, updating index name in the mapping file triggers HQ to
+create the new index with new mapping and reindex all data, so you'll
+have to update the index hash and alias at the top of the mapping file.
+The hash suffix to the index can just be a random alphanumeric string and
+is usually the date of the edit by convention. The alias should also be updated
+likewise or simply suffixed with ``_new``, for example ``xforms_new``. This will
+trigger a preindex as outlined in the `Indexes` section. In subsequent commits
+alias can be flipped back to what it was, for example ``xforms``. Changing the alias
+name doesn't trigger a reindex.
```

Comment:
I'm kind of just catching up here. Can you explain why we need to change the alias as well as the index name? I thought the whole point of using an alias and an index name was that the index name could change while keeping the alias the same

Reply:
OK sorry, I see the large comment above about this.

Looks like this also felt wrong to you guys but you couldn't figure out a way to use the index that still had the right behavior?

Reply:
Yeah, this doesn't feel completely right but is better than other options. This is discussed in [comments following this](https://github.com/dimagi/commcare-hq/pull/26733#issuecomment-610909316)

All of the current code uses index names directly rather than aliases, this PR starts making use of the aliases. But our index mapping change workflow includes building a new index with HQ itself, so without creating a separate alias during the index phase, there isn't any way to prevent writing to the old index itself.

Reply:
@dannyroberts I don't think it's that we couldn't figure out a way to use the index. An alias is designed to allow you to run searches across multiple indexes which is what we'd like to start using it for (e.g. monthly form indexes). We have been using it in CommCare as a way to 'flip' from one index to another after reindex (though we haven't actually used the aliases in code for a long time).